services/QuillLMS/engines/evidence/app/models/evidence/automl_model.rb

```diff
@@ -2,67 +2,70 @@
-#  automl_model_id :string           not null
-#  name            :string           not null
-#  labels          :string           default([]), is an Array
-#  prompt_id       :integer
-#  state           :string           not null
-#  created_at      :datetime         not null
-#  updated_at      :datetime         not null
-#  notes           :text             default("")
+#  id                   :bigint           not null, primary key
+#  labels               :string           default([]), is an Array
+#  name                 :string           not null
+#  notes                :text             default("")
+#  state                :string           not null
+#  created_at           :datetime         not null
+#  updated_at           :datetime         not null
+#  endpoint_external_id :string           not null
+#  model_external_id    :string           not null
+#  prompt_id            :bigint
+#
+# Indexes
+#
+#  index_evidence_automl_models_on_prompt_id  (prompt_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (prompt_id => comprehension_prompts.id)
 #
-require 'google/cloud/automl'
 
 module Evidence
   class AutomlModel < ApplicationRecord
-    self.table_name = 'comprehension_automl_models'
+    self.table_name = 'evidence_automl_models'
 
     include Evidence::ChangeLog
+
     MIN_LABELS_LENGTH = 1
+
     STATES = [
       STATE_ACTIVE = 'active',
       STATE_INACTIVE = 'inactive'
-    ]
-    PREDICT_API_TIMEOUT = 5.0
-    GOOGLE_PROJECT_ID = ENV['AUTOML_GOOGLE_PROJECT_ID']
-    GOOGLE_LOCATION = ENV['AUTOML_GOOGLE_LOCATION']
+    ].freeze
 
-    attr_readonly :automl_model_id, :name, :labels
+    attr_readonly :model_external_id, :endpoint_external_id, :name, :labels
 
     belongs_to :prompt, inverse_of: :automl_models
 
     validate :validate_label_associations, if: :active?
 
-    validates :automl_model_id, presence: true, uniqueness: true
+    validates :model_external_id, presence: true, uniqueness: true
+    validates :endpoint_external_id, presence: true, uniqueness: true
     validates :labels, presence: true, length: {minimum: MIN_LABELS_LENGTH}
     validates :name, presence: true
-    validates :state, inclusion: {in: ['active', 'inactive']}
-
-    before_validation :strip_whitespace
+    validates :state, inclusion: {in: STATES}
 
     def serializable_hash(options = nil)
       options ||= {}
 
       super(options.reverse_merge(
-        only: [:id, :automl_model_id, :notes, :name, :labels, :state, :prompt_id, :created_at, :updated_at],
+        only: [:id, :model_external_id, :endpoint_external_id, :notes, :name, :labels, :state, :prompt_id, :created_at, :updated_at],
         methods: [:older_models]
       ))
     end
 
-    def populate_from_automl_model_id
-      self.name = automl_name
-      self.labels = automl_labels
-      self.state = STATE_INACTIVE
+    def assign_custom_attributes
+      assign_attributes(VertexAI::ParamsBuilder.run(name).merge(state: STATE_INACTIVE))
```

Comment:
Take or leave (this currently follows the existing pattern): Consider moving this code to the controller `create` action since it's only used there and it deals will param parsing and building,.

Reply:
Good idea.  The current assignment felt weird in the model code.